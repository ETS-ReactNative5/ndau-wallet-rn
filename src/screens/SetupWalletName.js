import React, { Component } from 'react'
import {
  StyleSheet,
  View,
  ScrollView,
  Text,
  TextInput,
  Alert
} from 'react-native'
import CommonButton from '../components/CommonButton'
import SetupProgressBar from '../components/SetupProgressBar'
import cssStyles from '../css/styles'
import SetupStore from '../model/SetupStore'
import { SafeAreaView } from 'react-navigation'
import DataFormatHelper from '../helpers/DataFormatHelper'
import AsyncStorageHelper from '../model/AsyncStorageHelper'
import Padding from '../components/Padding'
import FlashNotification from '../components/FlashNotification'

class SetupEncryptionPassword extends Component {
  constructor (props) {
    super(props)

    this.state = {
      nameEntered: false
    }
    this.walletCount = 1
  }

  componentWillMount = () => {
    const user = this.props.navigation.getParam('user', null)
    if (user) {
      this.walletCount = Object.keys(user.wallets).length
    }
    SetupStore.walletId = `Wallet ${this.walletCount}`
  }

  checkIfWalletAlreadyExists = user => {
    if (
      DataFormatHelper.checkIfWalletAlreadyExists(user, SetupStore.walletId)
    ) {
      FlashNotification.showError(
        `There is already a wallet named "${SetupStore.walletId}". Please choose another name.`
      )
      return true
    }

    return false
  }

  showNextSetup = async () => {
    const { navigation } = this.props
    const user = navigation.getParam('user', null)

    if (this.checkIfWalletAlreadyExists(user)) {
      return
    }

    if (user) {
      // get it out of AppConstants.TEMP_USER if we have to
      // and into the new walletId
      DataFormatHelper.moveTempUserToWalletName(user, SetupStore.walletId)
    }

    // if we have an application password in
    // AsyncStorage then there is no need to show
    // this screen, so go to terms & conditions
    const password = await AsyncStorageHelper.getApplicationPassword()
    if (password) {
      SetupStore.encryptionPassword = password
      this.props.navigation.navigate('SetupTermsOfService', {
        user,
        walletSetupType: this.props.navigation.state.params &&
          this.props.navigation.state.params.walletSetupType
      })
    } else {
      if (user) {
        if (!user.userId) {
          user.userId = SetupStore.walletId
        }
        if (!user.wallets) {
          user.wallets = [{ walletId: SetupStore.walletId }]
        }
      }

      navigation.navigate('SetupEncryptionPassword', {
        user,
        walletSetupType: navigation.state.params &&
          navigation.state.params.walletSetupType
      })
    }
  }

  render () {
    return (
      <SafeAreaView style={cssStyles.safeContainer}>
        <View style={cssStyles.container}>
          <ScrollView style={cssStyles.contentContainer}>
            <SetupProgressBar navigation={this.props.navigation} />
            <Padding top={0}>
              <Text style={cssStyles.wizardText} onPress={this.showInformation}>
                Give this wallet a name.
              </Text>
            </Padding>

            <Padding>
              <TextInput
                style={cssStyles.textInput}
                onChangeText={value => {
                  this.setState({ nameEntered: true })
                  SetupStore.walletId = value
                }}
                // value={(value) => {
                //   SetupStore.walletId = value;
                // }}
                placeholder={`Wallet ${SetupStore.walletId}`}
                placeholderTextColor='#333'
                autoCapitalize='none'
              />
            </Padding>
          </ScrollView>
          <View style={cssStyles.footer}>
            <CommonButton
              onPress={this.showNextSetup}
              title='Next'
              // disabled={!this.state.nameEntered}
            />
          </View>
        </View>
      </SafeAreaView>
    )
  }
}

const styles = StyleSheet.create({
  textContainer: {
    marginBottom: 8
  },
  infoParagraph: {
    flexDirection: 'row'
  }
})

export default SetupEncryptionPassword
