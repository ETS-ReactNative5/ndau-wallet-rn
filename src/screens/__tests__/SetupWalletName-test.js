import { mount } from 'enzyme'
import React from 'react'
import { StyleSheet } from 'react-native'
import SetupWalletName from '../SetupWalletName'

import renderer from 'react-test-renderer'

describe('testing SetupWalletName...', () => {
  let styles = StyleSheet.create({
    wizardText: {
      color: '#ffffff',
      fontSize: 20
    }
  })
  const navigator = {
    setStyle: () => {},
    toggleNavBar: () => {}
  }

  it('renders correctly', () => {
    const tree = renderer
      .create(<SetupWalletName parentStyles={styles} navigator={navigator} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('can click the button', () => {
    const wrapper = mount(
      <SetupWalletName navigator={navigator} parentStyles={styles} />
    )
    const onlyButton = wrapper.find('Button').at(0)
    onlyButton.simulate('click')
  })
})
