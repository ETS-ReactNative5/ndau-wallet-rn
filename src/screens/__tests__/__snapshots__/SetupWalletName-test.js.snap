// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing SetupWalletName... renders correctly 1`] = `
<View
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "backgroundColor": "#1c2227",
      "flex": 1,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#1c2227",
        "flex": 1,
        "paddingBottom": 26.5,
        "paddingLeft": 15,
        "paddingRight": 15,
      }
    }
  >
    <RCTScrollView
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "marginBottom": 8,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#ffffff",
                "fontFamily": "TitilliumWeb-Regular",
                "fontSize": 20,
              }
            }
          >
            Give this wallet a name.
          </Text>
        </View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="Wallet name"
          placeholderTextColor="#333"
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderColor": "gray",
              "borderRadius": 3,
              "borderWidth": 1,
              "color": "#000000",
              "fontFamily": "TitilliumWeb-Regular",
              "fontSize": 18,
              "height": 93.5,
              "marginTop": 13.5,
              "paddingLeft": 7.5,
            }
          }
        />
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityTraits="button"
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "backgroundColor": "#696969",
            "borderRadius": 3,
            "height": 93.5,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "#ffffff",
              "fontFamily": "TitilliumWeb-Regular",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Next
        </Text>
      </View>
    </View>
  </View>
</View>
`;
